druid/src/widget/slider.rs

```diff
@@ -57,6 +59,24 @@ impl Slider {
         self
     }
 
+    /// Builder-style method to set the stepping.
+    ///
+    /// The default step size is `0.0` (smooth).
+    pub fn with_step(mut self, step: f64) -> Self {
+        if step < 0.0 {
+            warn!("bad stepping (must be positive): {}", step);
+            return self;
+        }
+        self.step = if step > 0.0 {
+            Some(step)
+        } else {
+            // A stepping value of 0.0 would yield an infinite amount of steps.
+            // Enforce no stepping instead.
+            None
+        };
```

Comment:
No need to change it, but just for posterity I'd have just written:
```suggestion
        if step < 0.0 {
            warn!("bad stepping (must be positive): {}", step);
        } else {
            self.step = Some(step);
        }
```

Reply:
That has a drawback IMO: you'd actually need to check for `step <= 0` since otherwise we would divide by zero elsewhere iirc.

We decided to allow `0` as a valid step value though, indicating smooth stepping (no intervals) - but we have to explicitly model this as `None`.

Reply:
lets change that `step > 0.0` to `step != 0.0` to make it more explicit